SplitIt/app/src/main/java/project/cs495/splitit/GroupViewActivity.java

```diff
@@ -0,0 +1,249 @@
+                    public void onClick(View view) {
+                        currGroupIndex = groupRV.getChildAdapterPosition(view);
+                        User member = (User) adapter.getItem(currGroupIndex);
+                    }
+                });
+
+                final ImageButton menu_options = view.findViewById(R.id.group_list_options);
+
+                // Use temporary variable to capture value of View
+                final View temp = view;
+                menu_options.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View view) {
+                        groupRV.findViewHolderForAdapterPosition(currGroupIndex).itemView.setSelected(false);
+                        currGroupIndex = groupRV.getChildAdapterPosition(temp);
+                        view.setSelected(true);
+                        PopupMenu popup = new PopupMenu(view.getContext(), view);
+                        popup.setOnMenuItemClickListener(GroupViewActivity.this);
+
+                        FirebaseAuth auth = FirebaseAuth.getInstance();
+                        User user = (User) adapter.getItem(currGroupIndex);
+                        if(auth.getCurrentUser().getUid().equals(user.getUid())) {
+                            MenuInflater inflater = popup.getMenuInflater();
+                            inflater.inflate(R.menu.manage_group_members_manager, popup.getMenu());
+                        }
+                        else {
+                            MenuInflater inflater = popup.getMenuInflater();
+                            inflater.inflate(R.menu.manage_group_members, popup.getMenu());
+                        }
+                        popup.show();
+                    }
+                });
+
+                return new memberHolder(view);
+            }
+        };
+        groupRV.setAdapter(adapter);
+        groupRV.setLayoutManager(new LinearLayoutManager(this));
+
+        fab_plus = (ImageButton) findViewById(R.id.add_member);
+        fab_plus.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View view) {
+                openDialog(groupId);
+            }
+        });
+    }
+
+    public void openDialog(String groupId) {
+        MemberAddDialog memberAddDialog = new MemberAddDialog(this,groupId);
+        memberAddDialog.show();
+    }
+
+    public void leaveGroup(String groupId) {
+        FirebaseAuth auth = FirebaseAuth.getInstance();
+        if (auth.getCurrentUser().getUid().equals(group.getManagerUID()))
+            Toast.makeText(GroupViewActivity.this, getString(R.string.owner_leave), Toast.LENGTH_SHORT).show();
+        else {
+            Utils.getDatabaseReference().child("groups").child(groupId).child("memberID").child(auth.getCurrentUser().getUid()).removeValue();
+            Utils.getDatabaseReference().child("groups").child(groupId).child("members").child(auth.getCurrentUser().getDisplayName()).removeValue();
```

Comment:
I think we should consider getting rid of the members field altogether.  Any time we need the display name we really should be getting it from the user object, as the two list do not maintain the same ordering.